go/private/rules/transition.bzl

```diff
@@ -117,15 +120,13 @@ def _go_transition_impl(settings, attr):
 
     tags = getattr(attr, "gotags", [])
     if tags:
-        tags_label = "//go/config:tags"
-        settings[tags_label] = tags
+        settings["//go/config:tags"] = _deduped_and_sorted(tags)
```

Comment:
you can use `sets.to_list(sets.make(tags))` from skylib: https://github.com/bazelbuild/bazel-skylib/blob/main/docs/new_sets_doc.md

Reply:
That would still require a call to `sorted` though, so I'm not sure it would reduce the overall complexity here.